Added support for Pimoroni Display Hat Mini

makerph opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Added support for Pimoroni Display Hat Mini
- 2.0” 320x240 pixel IPS LCD screen, connected via SPI (~220 PPI, 65K colours)
[
![image](https://user-images.githubusercontent.com/13675709/184265013-5eb75a81-d829-4777-b924-f13959756cb9.png)
](url)

![A8DE980D-1F6F-4066-AA80-D0A24DD75388](https://user-images.githubusercontent.com/13675709/184265645-49176fda-f9c0-48df-8c51-3401fdacfc7d.jpg)


## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
I wanted to have a large display with larger resolution in phat format. There are existing LCD Hat support but text is too small for my old eyes.

- [ ] I have raised an issue to propose this change ([required](https://github.com/evilsocket/pwnagotchi/blob/master/CONTRIBUTING.md))


## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested this on Raspberry Pi Zero W with fresh 1.5.5 Image

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/evilsocket/pwnagotchi/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`

do-ki committed:
Add Display Hat Support

Add Pimoroni Display Hat support

do-ki committed:
Add Pimoroni Display Hat Support

Add Pimoroni Display Hat Support

do-ki committed:
add display hat mini support

add pimoroni display hat mini support

do-ki committed:
Create ST7789.py

Add Pimoroni Display Hat Mini Support

do-ki committed:
Update __init__.py

Add Pimoroni Hat Support

do-ki committed:
Update displayhatmini.py

fix class error

do-ki committed:
Update ST7789.py

change screen width, height and rotation

do-ki committed:
Update ST7789.py

fix: added backlight pin

do-ki committed:
Update displayhatmini.py

fix: added backlight pin

do-ki committed:
Update displayhatmini.py

cleanup

do-ki committed:
Update __init__.py

fix missing lines

do-ki committed:
Update ST7789.py

change spi speed

do-ki committed:
Update displayhatmini.py

update logging info message

DizHell commented:
Merci @do-ki  Testé ce jour et c'est fonctionnel.
Je vais regarder pour faire des mofications au niveau de la luminausité, et regler l'affichage des stats CPU MEM...
Suivra ensuite l'ajout de la prise en compte des boutons et Led RGB.

Thanks @do-ki Tested today and it's functional.
I'm going to look to make changes to the brightness, and adjust the display of CPU MEM stats...
This will then be followed by the addition of support for buttons and RGB LEDs.
<img width="1372" alt="image" src="https://github.com/evilsocket/pwnagotchi/assets/33041518/0edaa891-1dc0-48a3-87ae-b53071e16000">

do-ki committed:
Update net-pos.py

- Fixed self.skip += np_file to self.skip.append(np_file)
- Replaced unused _append_saved() with in-use logic
- Made .net_pos_saved file path configurable
- Updated API_URL to use the Google Geolocation API
- Cleaned up logging and removed unused code
